lib/handler/mruby.c

```diff
@@ -333,6 +363,8 @@ static void on_context_init(h2o_handler_t *_handler, h2o_context_t *ctx)
     /* compile code (must be done for each thread) */
     int arena = mrb_gc_arena_save(handler_ctx->mrb);
     mrb_value proc = h2o_mruby_compile_code(handler_ctx->mrb, &handler->config, NULL);
+    validate_proc(handler_ctx->mrb, proc);
```

Comment:
Thank you for implementing the check in the way I suggested in our discussion.

I am sorry to say this, but after reading this, I think we should better let the mruby script _set_ an optional callback for validating the Rack handler, instead of having a hard-coded check implemented in the C side. WDYT?